accounting/invoice/admin.py

```diff
@@ -21,30 +21,83 @@
 Administration for Invoices.
 """
 
+from django.conf import settings
 from django.contrib import admin
+from django.utils.html import format_html
+from rangefilter.filter import DateRangeFilter
 from simple_history.admin import SimpleHistoryAdmin
 
 from accounting.common.admin import UuidModelAdmin
-from accounting.invoice import models
+from accounting.invoice import choices, models
 
 
 @admin.register(models.InvoiceTemplate)
 class InvoiceTemplateAdmin(UuidModelAdmin):
     """ Admin configuration for the `InvoiceTemplate` model. """
+
     list_display = UuidModelAdmin.list_display + ('provider', 'numbering_scheme',)
     search_fields = ('provider__user__username',)
 
 
+# pylint: disable=no-self-use
 @admin.register(models.Invoice)
 class InvoiceAdmin(UuidModelAdmin, SimpleHistoryAdmin):
     """ Admin configuration for the `Invoice` model. """
-    list_display = UuidModelAdmin.list_display + ('number', 'provider', 'client', 'due_date', 'approved', 'paid',)
+
+    list_display = ('__str__', 'provider', 'client', 'total', 'pdf', 'jira_timesheet', 'approved', 'paid',)
+    list_filter = (('date', DateRangeFilter),)
     search_fields = ('provider__user__username', 'client__user__username',)
-    readonly_fields = ('pdf_path',)
+    readonly_fields = ('pdf', 'jira_timesheet', 'total',)
+    actions = ['mark_paid', 'mark_approved']
+    exclude = ('pdf_path',)
+
+    def total(self, instance):
+        """The total charge of this invoice, given in the currency of the hourly rate between provider and client."""
+        total_cost = instance.total_cost or 0
```

Comment:
@itsjeyd When there aren't any line items, whatever value `total_cost` was returning was causing tracebacks when converting to a string with `0.2f` format I think. So as a safety I defaulted to an actual number.

Reply:
@UmanShahzad OK, it might be worth considering an update to `total_cost` (and perhaps also `total_quantity`) to make sure it always returns a number. Making client code responsible for that seems a bit odd. If you agree, feel free to address this as part of @antoviaque's review, though.

Reply:
@itsjeyd That actually makes sense, yeah. I'll do that when @antoviaque does a review, sure.

Reply:
+1